src/H5CX.c

```diff
@@ -931,33 +931,27 @@ H5CX_retrieve_state(H5CX_state_t **api_state)
     } /* end if */
 
     /* Keep a copy of the VOL connector property, if there is one */
-    if ((*head)->ctx.vol_connector_prop_valid && (*head)->ctx.vol_connector_prop.connector_id > 0) {
+    if ((*head)->ctx.vol_connector_prop_valid && (*head)->ctx.vol_connector_prop.connector) {
```

Comment:
VOL connector property stores a pointer to the connector struct now, instead of nesting an ID inside the property list.